src/output.cpp

```diff
@@ -698,24 +681,19 @@ std::string JsonOutput::hist_to_str(const std::vector<uint64_t> &values,
       res << ", ";
 
     res << "{";
-    if (i == 0)
-    {
-      res << "\"max\": -1, ";
-    }
-    else if (i == 1)
-    {
-      res << "\"min\": 0, \"max\": 0, ";
-    }
-    else if (i == 2)
+    long low = i - 1, high = i - 1;  // covers negative and first 2<<bits slots
+    if (i > (2 << bits))
     {
-      res << "\"min\": 1, \"max\": 1, ";
-    }
-    else
-    {
-      long low = 1ULL << (i - 2);
-      long high = (1ULL << (i - 2 + 1)) - 1;
-      res << "\"min\": " << low << ", \"max\": " << high << ", ";
+      uint32_t power = (i - 1) >> bits, n = 1<<bits, frac = (i - 1) & (n - 1);
+      low = (1ul << (power - 1) ) * (n + frac);
+      power = i >> bits;
+      frac = i & (n - 1);
+      high = (1ul << (power - 1)) * (n + frac);
```

Comment:
need to fix this. the text output prints open intervals [low,high), json uses closed intervals so max should be high-1

Reply:
fixed and verified tests pass.